test/correctness/autodiff.cpp

```diff
@@ -1246,7 +1246,8 @@ void test_select_guard() {
 }
 
 void test_param() {
-    Param<float> param(2.f);
+    // Give the parameter an illegal name to test the function renaming.
+    Param<float> param("param.0", 2.f);
```

Comment:
We may forbid explicit param names with dots at some point. Instead, make it a buffer param, and use its width where you use param below.

Reply:
Like this? https://github.com/halide/Halide/pull/4016/commits/bac5e9be0e113b65cecce2cb4a01d02d5223356b

Reply:
Accidentally turned out all other tests. Fixed in https://github.com/halide/Halide/pull/4016/commits/1e4076d653b7bc5459b99a3205efde923d0acf9d

Reply:
Sorry, my brain wasn't working. Fixed the test by using an ImageParam. See https://github.com/halide/Halide/pull/4016/commits/e7ca5649a079d66ab72a24af582cfe76ed3aaa7c